# 📊 Parcial – Análisis de Redes Sociales (Primer Corte)

## 🎯 Objetivo
Evaluar tu dominio de **filtros** y **agrupaciones** en Python (sin pivots ni gráficos) y tu capacidad de **análisis de negocio** en métricas de marketing: **CAC, LTV, churn rate, funnel, unit economics (LTV/CAC)**.

---
## 🧩 Contexto de Negocio (Caso Real)
**StartUp SaaS en crecimiento.** Vende planes **Basic, Pro, Enterprise** en **LatAm, North America, Europe, APAC**. Los canales de adquisición son: **meta_ads, google_ads, organic_search, outbound_sales**. Los usuarios se registran por **web** o **mobile_app**.

La dirección quiere decidir **dónde invertir el presupuesto del próximo trimestre**. Te piden:

1) Identificar **canales más rentables** (relación **LTV/CAC** y churn).  
2) Detectar **regiones** con mayor **potencial de crecimiento** y/o **riesgo**.  
3) Evaluar el desempeño por **tier** (Basic, Pro, Enterprise).  
4) Recomendar **asignación de presupuesto** por **canal + tier**.

---
## 📂 Instrucciones
- Trabaja únicamente con **filtros** y **agrupaciones**. **No uses** tablas dinámicas (pivot), merges ni gráficos.
- Archivo a usar: `clientes_marketing.csv`.
- Escribe **código + interpretación breve** para cada inciso.
- Si el enunciado dice **“Agrupa”**, usa `groupby`. Si dice **“Filtra”**, usa filtrado con máscaras booleanas. Si dice **“Ambas”**, realiza primero el **filtro** y luego la **agrupación**.

---


## Descargar Tabla (Correr una vez)

In [5]:
!wget https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/cac_ltv_model.csv

--2025-09-17 23:57:28--  https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/cac_ltv_model.csv
Resolving github.com (github.com)... 140.82.116.3
Connecting to github.com (github.com)|140.82.116.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/cac_ltv_model.csv [following]
--2025-09-17 23:57:28--  https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/cac_ltv_model.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 661398 (646K) [text/plain]
Saving to: ‘cac_ltv_model.csv’

cac_ltv_model.csv   100%[===================>] 645.90K  --.-KB/s

## Importar Datos

In [69]:
import pandas as pd
pd.set_option('display.float_format', lambda x: f'{x:,.2f}')

# 1) Cargar dataset
data = pd.read_csv('cac_ltv_model.csv')


### ✅ Variables relevantes (recordatorio)
- `acquisition_channel`, `signup_source`, `region`, `customer_tier`
- `plan_price`, `discount_rate`, `arpu`, `gross_margin`, `churn_rate`
- `contract_length_months`, `marketing_spend`

### 🧮 Fórmulas de negocio (a usar en el examen)
- **CAC por canal**:  
    $$ CAC = \frac{\text{Total Marketing Spend}}{\text{Clientes adquiridos}} $$
  
- **LTV por observación** (aprox.):  
   $$ LTV = ARPU \times contract\_length\_months \times gross\_margin \times (1 - churn\_rate) $$

- **Unit economics**:  
  $$ LTV/CAC   $$(ideal \> 3 como regla general de SaaS)

---



### Parte Exploratoria: Cree una muestra de los primeros 5 elementos de esta tabla. ¿Cuantas observaciones tienes?

In [7]:
data.head()

year  month    date  customer_id acquisition_channel signup_source  \
0  2023      1  Jan-23         1001      outbound_sales           web   
1  2023      1  Jan-23         1002            meta_ads           web   
2  2023      1  Jan-23         1003      organic_search    mobile_app   
3  2023      1  Jan-23         1004      organic_search           web   
4  2023      1  Jan-23         1005      organic_search           web   

          region customer_tier  plan_price  discount_rate   arpu  \
0  North America         Basic       78.84           0.10  63.63   
1          LatAm           Pro      120.23           0.00 124.47   
2  North America    Enterprise      335.82           0.20 278.20   
3         Europe           Pro      193.89           0.10 190.65   
4           APAC    Enterprise      471.80           0.10 445.32   

   gross_margin  churn_rate  contract_length_months  marketing_spend  
0          0.76        0.02                      12           212.48  
1          0.84        0.05                       1           352.60  
2          0.81        0.02                      12            55.40  
3          0.80        0.08                       1            49.10  
4          0.80        0.08                       1            45.27

## 🔎 Parte A – Filtros

hint: Crea una tabla con el filtro y luego agrupa la variable que te piden.

1. **Filtra** los clientes del canal **`meta_ads`**. Calcula el **ARPU promedio** de este subconjunto.  



In [37]:
Filtro = data.loc[data["acquisition_channel"]== "meta_ads"]

In [54]:
print(Filtro)

      year  month    date  customer_id acquisition_channel signup_source  \
1     2023      1  Jan-23         1002            meta_ads           web   
10    2023      1  Jan-23         1011            meta_ads           web   
15    2023      1  Jan-23         1016            meta_ads    mobile_app   
19    2023      1  Jan-23         1020            meta_ads    mobile_app   
23    2023      1  Jan-23         1024            meta_ads    mobile_app   
...    ...    ...     ...          ...                 ...           ...   
7044  2024     12  Dec-24         8045            meta_ads       partner   
7048  2024     12  Dec-24         8049            meta_ads           web   
7051  2024     12  Dec-24         8052            meta_ads           web   
7053  2024     12  Dec-24         8054            meta_ads    mobile_app   
7055  2024     12  Dec-24         8056            meta_ads           web   

             region customer_tier  plan_price  discount_rate   arpu  \
1             La

In [52]:
ARPU = Filtro["arpu"].mean()

In [53]:
print(ARPU)

168.78858272162617


2. **Filtra** clientes de **LatAm** con **`churn_rate > 0.05`**. ¿Cuántos son? ¿Cuál es su **ARPU promedio**?  

In [32]:
Filtro1 = data.loc[(data["region"]== "LatAm") & (data["churn_rate"]> 0.05)]

In [48]:
print(Filtro1)

      year  month    date  customer_id acquisition_channel signup_source  \
1     2023      1  Jan-23         1002            meta_ads           web   
25    2023      1  Jan-23         1026      outbound_sales           web   
49    2023      1  Jan-23         1050            meta_ads           web   
65    2023      1  Jan-23         1066      organic_search       partner   
92    2023      1  Jan-23         1093          google_ads       partner   
...    ...    ...     ...          ...                 ...           ...   
6967  2024     12  Dec-24         7968      organic_search    mobile_app   
6976  2024     12  Dec-24         7977      outbound_sales           web   
7015  2024     12  Dec-24         8016      outbound_sales    mobile_app   
7017  2024     12  Dec-24         8018          google_ads           web   
7046  2024     12  Dec-24         8047          google_ads    mobile_app   

     region customer_tier  plan_price  discount_rate   arpu  gross_margin  \
1     LatA

In [49]:
ARPU1 = Filtro1["arpu"].mean()

In [50]:
print(ARPU1)

174.78453658536586


3. **Filtra** clientes **Enterprise** con `contract_length_months > 6`. ¿Cuál es su **gross_margin promedio**?

In [73]:
Filtro2 = data.loc[(data["customer_tier"]== "Enterprise") & (data["contract_length_months"]> 6)]

In [74]:
GroosMargin = Filtro2["gross_margin"].mean()

In [75]:
print(GroosMargin)

0.8248397435897435


## 📊 Parte B – Agrupaciones (SOLO agrupar)
4. **Agrupa** por `acquisition_channel` y calcula **ARPU promedio**. Ordena de mayor a menor.  


In [64]:
data.groupby("acquisition_channel")["arpu"].mean().sort_values(ascending=False)

acquisition_channel
outbound_sales   173.61
google_ads       172.37
organic_search   171.14
meta_ads         168.79
Name: arpu, dtype: float64

5. **Agrupa** por `region` y calcula **churn_rate promedio**. Identifica la región con mayor churn.  



In [65]:
data.groupby("region")["churn_rate"].mean().sort_values(ascending=False)

region
Middle East     0.05
LatAm           0.05
North America   0.05
APAC            0.05
Europe          0.05
Africa          0.05
Name: churn_rate, dtype: float64

6. **Agrupa** por `customer_tier` y calcula **marketing_spend total**. ¿Cuál tier consume más presupuesto?

In [67]:
data.groupby("customer_tier")["marketing_spend"].sum().sort_values(ascending=False)

customer_tier
Pro          589,265.04
Basic        584,419.11
Enterprise   584,348.87
Name: marketing_spend, dtype: float64

## 🔀 Parte C – Filtro **y** Agrupación (Ambas)
7. **Filtra** solo registros de **`signup_source = 'web'`** y luego **agrupa** por `acquisition_channel` para obtener el **churn_rate promedio**.  


In [83]:
Filtro3 = data.loc[(data["signup_source"]== "web") & (data["acquisition_channel"])]

In [76]:
churn_rate1 = Filtro3["churn_rate"].mean()

In [77]:
print(churn_rate1)

0.053168505263157896


8. **Filtra** solo **`mobile_app`** y luego **agrupa** por `region` para calcular **ARPU promedio**.  


In [78]:
Filtro4 = data.loc[(data["signup_source"]== "mobile_app") & (data["region"])]

In [79]:
ARPU8 = Filtro4["arpu"].mean()

In [80]:
print(ARPU8)

170.48253737718923


9. **Filtra** a clientes **`Pro`** y **agrupa** por `acquisition_channel` para calcular **marketing_spend total**.


In [82]:
Filtro5 = data.loc[(data["customer_tier"]== "Pro") & (data["acquisition_channel"])]

In [84]:
marketing_spendtotal1 = Filtro5['marketing_spend'].sum()

In [85]:
print(marketing_spendtotal1)

589265.04


## 📈 Parte D – Métricas de negocio (CAC, LTV, LTV/CAC)
10. **CAC por canal (Agrupa)**: calcula el CAC de cada `acquisition_channel` como:  
   `CAC = marketing_spend_total_del_canal / #clientes_del_canal`  


In [86]:
data.head()

year  month    date  customer_id acquisition_channel signup_source  \
0  2023      1  Jan-23         1001      outbound_sales           web   
1  2023      1  Jan-23         1002            meta_ads           web   
2  2023      1  Jan-23         1003      organic_search    mobile_app   
3  2023      1  Jan-23         1004      organic_search           web   
4  2023      1  Jan-23         1005      organic_search           web   

          region customer_tier  plan_price  discount_rate   arpu  \
0  North America         Basic       78.84           0.10  63.63   
1          LatAm           Pro      120.23           0.00 124.47   
2  North America    Enterprise      335.82           0.20 278.20   
3         Europe           Pro      193.89           0.10 190.65   
4           APAC    Enterprise      471.80           0.10 445.32   

   gross_margin  churn_rate  contract_length_months  marketing_spend  
0          0.76        0.02                      12           212.48  
1          0.84        0.05                       1           352.60  
2          0.81        0.02                      12            55.40  
3          0.80        0.08                       1            49.10  
4          0.80        0.08                       1            45.27

In [114]:
marketing_spend_total_del_canal = data.groupby("acquisition_channel")["marketing_spend"].sum()

In [112]:
numeroclientes_del_canal = data["acquisition_channel"].value_counts()

In [115]:
print(marketing_spend_total_del_canal)

acquisition_channel
google_ads       696,163.32
meta_ads         618,560.13
organic_search    88,917.87
outbound_sales   354,391.70
Name: marketing_spend, dtype: float64


In [96]:
print(numeroclientes_del_canal)

acquisition_channel
organic_search    1780
meta_ads          1771
outbound_sales    1770
google_ads        1736
Name: count, dtype: int64


In [119]:
CACporcanal = marketing_spend_total_del_canal/numeroclientes_del_canal

In [117]:
marketing_spend_total_del_canal/numeroclientes_del_canal

acquisition_channel
google_ads       401.02
meta_ads         349.27
organic_search    49.95
outbound_sales   200.22
dtype: float64

11. **CAC por tier (Agrupa)**: calcula el CAC de cada `customer_tier` como:  
   `CAC = marketing_spend_total_del_tier / #clientes_del_canal`  


In [101]:
marketing_spend_total_del_tier = data.groupby("customer_tier")["marketing_spend"].sum()


In [108]:
numeroclientes_del_canal11 = data["customer_tier"].value_counts()

In [106]:
print(marketing_spend_total_del_tier)

customer_tier
Basic        584,419.11
Enterprise   584,348.87
Pro          589,265.04
Name: marketing_spend, dtype: float64


In [109]:
print(numeroclientes_del_canal11)

customer_tier
Pro           2372
Enterprise    2345
Basic         2340
Name: count, dtype: int64


In [120]:
CACtier = marketing_spend_total_del_tier/numeroclientes_del_canal11

12. **LTV por tier (Ambas)**: crea una columna `ltv_individual` con la fórmula dada.


In [134]:
data['ltv_individual'] = data['arpu'] * data['contract_length_months'] * data['gross_margin'] * (1 - data['churn_rate'])
display(data.head())

year  month    date  customer_id acquisition_channel signup_source  \
0  2023      1  Jan-23         1001      outbound_sales           web   
1  2023      1  Jan-23         1002            meta_ads           web   
2  2023      1  Jan-23         1003      organic_search    mobile_app   
3  2023      1  Jan-23         1004      organic_search           web   
4  2023      1  Jan-23         1005      organic_search           web   

          region customer_tier  plan_price  discount_rate   arpu  \
0  North America         Basic       78.84           0.10  63.63   
1          LatAm           Pro      120.23           0.00 124.47   
2  North America    Enterprise      335.82           0.20 278.20   
3         Europe           Pro      193.89           0.10 190.65   
4           APAC    Enterprise      471.80           0.10 445.32   

   gross_margin  churn_rate  contract_length_months  marketing_spend  \
0          0.76        0.02                      12           212.48   
1          0.84        0.05                       1           352.60   
2          0.81        0.02                      12            55.40   
3          0.80        0.08                       1            49.10   
4          0.80        0.08                       1            45.27   

   ltv_individual  
0          567.54  
1           99.20  
2        2,651.91  
3          139.68  
4          328.72

12. A. Luego **agrupa** `ltv_individual` por
`acquisition_channel` para obtener el **LTV promedio**.  

In [135]:
ltv_por_canal = data.groupby('acquisition_channel')['ltv_individual'].mean()
print("LTV promedio por canal de adquisición:")
print(ltv_por_canal)

LTV promedio por canal de adquisición:
acquisition_channel
google_ads       710.25
meta_ads         729.05
organic_search   765.13
outbound_sales   751.57
Name: ltv_individual, dtype: float64


12. B. Luego **agrupa** `ltv_individual`por `customer_tier` para obtener el **LTV promedio**.





In [136]:
ltv_por_tier = data.groupby('customer_tier')['ltv_individual'].mean()
print("\nLTV promedio por nivel de cliente:")
print(ltv_por_tier)


LTV promedio por nivel de cliente:
customer_tier
Basic          291.77
Enterprise   1,349.02
Pro            577.63
Name: ltv_individual, dtype: float64


13. **Unit economics (Ambas)**: combina tus resultados para comparar **LTV promedio por tier** contra **CAC por canal** y comenta **qué combinaciones canal + tier** lucen más saludables (busca **LTV/CAC > 3**).


In [137]:
# Group by acquisition_channel and customer_tier to calculate total marketing_spend and average ltv_individual
ltv_cac_analysis = data.groupby(['acquisition_channel', 'customer_tier']).agg(
    total_marketing_spend=('marketing_spend', 'sum'),
    num_customers=('customer_id', 'count'),
    avg_ltv=('ltv_individual', 'mean')
)

# Calculate CAC for each group
ltv_cac_analysis['cac'] = ltv_cac_analysis['total_marketing_spend'] / ltv_cac_analysis['num_customers']

# Calculate LTV/CAC ratio
ltv_cac_analysis['ltv_cac_ratio'] = ltv_cac_analysis['avg_ltv'] / ltv_cac_analysis['cac']

# Display the results, sorted by LTV/CAC ratio
display(ltv_cac_analysis.sort_values(by='ltv_cac_ratio', ascending=False))

total_marketing_spend  num_customers  \
acquisition_channel customer_tier                                         
organic_search      Enterprise                 29,633.76            592   
                    Pro                        29,848.72            597   
outbound_sales      Enterprise                117,786.78            589   
organic_search      Basic                      29,435.39            591   
meta_ads            Enterprise                195,857.06            564   
google_ads          Enterprise                241,071.27            600   
outbound_sales      Pro                       121,177.04            605   
meta_ads            Pro                       211,517.60            605   
outbound_sales      Basic                     115,427.88            576   
google_ads          Pro                       226,721.68            565   
meta_ads            Basic                     211,185.47            602   
google_ads          Basic                     228,370.37            571   

                                   avg_ltv    cac  ltv_cac_ratio  
acquisition_channel customer_tier                                 
organic_search      Enterprise    1,410.71  50.06          28.18  
                    Pro             596.45  50.00          11.93  
outbound_sales      Enterprise    1,361.48 199.98           6.81  
organic_search      Basic           288.84  49.81           5.80  
meta_ads            Enterprise    1,353.77 347.26           3.90  
google_ads          Enterprise    1,271.47 401.79           3.16  
outbound_sales      Pro             594.23 200.29           2.97  
meta_ads            Pro             558.70 349.62           1.60  
outbound_sales      Basic           293.15 200.40           1.46  
google_ads          Pro             560.24 401.28           1.40  
meta_ads            Basic           314.96 350.81           0.90  
google_ads          Basic           268.95 399.95           0.67

In [138]:
# Filter for LTV/CAC ratio > 3
healthy_combinations = ltv_cac_analysis[ltv_cac_analysis['ltv_cac_ratio'] > 3]

# Display the filtered results
print("Combinaciones Canal + Tier con LTV/CAC > 3:")
display(healthy_combinations)

Combinaciones Canal + Tier con LTV/CAC > 3:


total_marketing_spend  num_customers  \
acquisition_channel customer_tier                                         
google_ads          Enterprise                241,071.27            600   
meta_ads            Enterprise                195,857.06            564   
organic_search      Basic                      29,435.39            591   
                    Enterprise                 29,633.76            592   
                    Pro                        29,848.72            597   
outbound_sales      Enterprise                117,786.78            589   

                                   avg_ltv    cac  ltv_cac_ratio  
acquisition_channel customer_tier                                 
google_ads          Enterprise    1,271.47 401.79           3.16  
meta_ads            Enterprise    1,353.77 347.26           3.90  
organic_search      Basic           288.84  49.81           5.80  
                    Enterprise    1,410.71  50.06          28.18  
                    Pro             596.45  50.00          11.93  
outbound_sales      Enterprise    1,361.48 199.98           6.81

## 🧠 Parte E – Análisis crítico (respuesta abierta)
13. **CMO por un día**: con tus resultados, ¿en qué **canal(es)** invertirías más el próximo trimestre? ¿Por qué? (Cita **CAC** y **churn**).  
14. ¿Qué **región** ves con mayor riesgo? ¿Qué hipótesis explicarían su **churn**?  
15. Identifica **dos combinaciones canal + tier** con mejor **LTV/CAC** y propón **dos acciones** concretas de optimización (p. ej., creatividades, audiencias, pricing, retención).

---
✍️ **Nota**: Mantente disciplinado con el enunciado (**Filtra**, **Agrupa**, **Ambas**). La evaluación pondera **correctitud técnica** y **calidad de interpretación de negocio**.


🧠 Parte E – Análisis crítico (respuesta abierta)
13. CMO por un día: con tus resultados, ¿en qué canal(es) invertirías más el próximo trimestre? ¿Por qué? (Cita CAC y churn).

Basándome en los resultados, invertiría más en el canal Organic Search, especialmente para los tiers Enterprise y Pro.

Organic Search tiene el CAC más bajo (49.95) en comparación con los otros canales (meta_ads: 349.27, google_ads: 401.02, outbound_sales: 200.22). Esto significa que adquirir un cliente a través de búsqueda orgánica es significativamente más económico.
Aunque el churn rate promedio por región es similar (alrededor de 0.05), al observar las combinaciones de canal y tier, vemos que Organic Search para Enterprise y Pro tiene LTV/CAC ratios excepcionalmente altos (28.18 y 11.93 respectivamente), lo que indica una alta rentabilidad y retención en estos segmentos.


14. ¿Qué región ves con mayor riesgo? ¿Qué hipótesis explicarían su churn?

Aunque el churn rate promedio por región parece similar, sería importante analizar el churn por combinación de región y tier, o por región y canal para identificar riesgos específicos. Si, por ejemplo, una región mostrara un churn significativamente más alto para un tier o canal específico, esa sería una señal de alerta.

Hipótesis que explicarían un churn alto en una región podrían ser:

Competencia local intensa: Otras empresas  en esa región ofrecen mejores alternativas o precios más bajos.
Adaptación del producto/servicio: El producto o servicio no se ajusta completamente a las necesidades o preferencias de los clientes en esa región (idioma, características culturales, métodos de pago, etc.).
Problemas de soporte o atención al cliente: La calidad del soporte o la atención al cliente en esa región no es adecuada, lo que lleva a la frustración y cancelación.
Condiciones económicas: Una economía inestable o recesión en la región puede llevar a los clientes a reducir gastos.
Regulaciones locales: Regulaciones específicas de la región que dificultan el uso del servicio o aumentan los costos para el cliente.
Para confirmar estas hipótesis, sería necesario realizar análisis más detallados, encuestas a clientes y estudios de mercado en la región identificada.

15. Identifica dos combinaciones canal + tier con mejor LTV/CAC y propón dos acciones concretas de optimización (p. ej., creatividades, audiencias, pricing, retención).

Dos combinaciones canal + tier con mejor LTV/CAC son:

Organic Search - Enterprise (LTV/CAC: 28.18)
Organic Search - Pro (LTV/CAC: 11.93)
Ambas combinaciones provienen del canal Organic Search, que ya es muy eficiente en términos de CAC. Las acciones de optimización se centrarían en maximizar el valor de estos clientes y atraer a más clientes similares:

Acción de Optimización 1 (Organic Search - Enterprise): Mejorar la retención y upselling. Dado el alto LTV, estos clientes ya son muy valiosos. Podríamos implementar un programa de customer success dedicado para clientes Enterprise, ofreciendo soporte proactivo, capacitaciones avanzadas y revisando periódicamente sus necesidades para identificar oportunidades de upselling a funcionalidades o servicios premium. Esto aseguraría que sigan obteniendo valor del producto y reduciría aún más el churn.
Acción de Optimización 2 (Organic Search - Pro): Optimizar el funnel de conversión en la web/mobile app. Dado el buen LTV/CAC, el canal orgánico ya atrae a clientes Pro rentables. La optimización podría enfocarse en mejorar las páginas de destino para usuarios que llegan por búsqueda orgánica, asegurando que la propuesta de valor para el tier Pro sea clara, que el proceso de registro sea fluido y que se destaquen los beneficios clave para este segmento. Pruebas A/B en la web y la app podrían ayudar a identificar los elementos más efectivos para aumentar la tasa de conversión de visitantes orgánicos a clientes Pro.